tests/test_finosmeetings.py

```diff
@@ -0,0 +1,218 @@
+        http_requests.append(last_request)
+
+        return (200, headers, body)
+
+    httpretty.register_uri(httpretty.GET,
+                           MEETINGS_URL,
+                           responses=[
+                               httpretty.Response(body=request_callback)
+                               for _ in range(2)
+                           ])
+
+    return http_requests
+
+
+class TestFinosMeetingsBackend(unittest.TestCase):
+    """FinosMeetings backend tests"""
+
+    def test_initialization(self):
+        """Test whether attributes are initializated"""
+
+        finosmeetings = FinosMeetings(
+            MEETINGS_URL,
+            csv_header='email,name,org,githubid,program,activity,date',
+            separator=',',
+            date_formats='%a %b %d %H:%M:%S EDT %Y, %Y-%m-%d,%Y-%m-%d,%Y-%m,%Y',
+            skip_header='true',
+            id_columns='email,name,date',
+            date_column='date')
+
+        self.assertTrue(finosmeetings.file_path, MEETINGS_URL)
+        self.assertEqual(finosmeetings.csv_header, 'email,name,org,githubid,program,activity,date')
+        self.assertEqual(finosmeetings.separator, ',')
+        self.assertEqual(finosmeetings.date_formats, '%a %b %d %H:%M:%S EDT %Y, %Y-%m-%d,%Y-%m-%d,%Y-%m,%Y')
+        self.assertEqual(finosmeetings.skip_header, True)
+        self.assertEqual(finosmeetings.id_columns, 'email,name,date')
+        self.assertEqual(finosmeetings.date_column, 'date')
+        self.assertIsNone(finosmeetings.client)
+
+    def test_has_archiving(self):
+        """Test if it returns True when has_archiving is called"""
+
+        self.assertEqual(FinosMeetings.has_archiving(), False)
+
+    def test_has_resuming(self):
+        """Test if it returns False when has_resuming is called"""
+
+        self.assertEqual(FinosMeetings.has_resuming(), False)
+
+    @httpretty.activate
+    def test_fetch(self):
+        """Test whether a list of entries is returned"""
+
+        http_requests = configure_http_server()
+
+        # Test fetch entries from feed
+        finosmeetings = FinosMeetings(
+            MEETINGS_URL,
+            csv_header='email,name,org,githubid,program,activity,date',
+            separator=',',
+            date_formats='%a %b %d %H:%M:%S EDT %Y, %Y-%m-%d,%Y-%m-%d,%Y-%m,%Y',
+            skip_header='true',
+            id_columns='email,name,date',
+            date_column='date')
+
+        entries = [entry for entry in finosmeetings.fetch()]
+        self.assertEqual(len(entries), 3)
+        self.assertEqual(len(http_requests), 1)
+
+        # Test metadata
+        expected = [('rob.underwood@finos.org', 'Rob Underwood', 'brooklynrob', 'Data Tech', 'Data Tech PMC', '2018-09-28'),
+                    ('tosha.ellison@finos.org', 'Tosha Ellison', '', 'Data Tech', 'Security Reference Data', '2018-12-11'),
+                    ('maoo@finos.org', 'Maurizio Pillitu', 'maoo', 'FDC3', 'FDC3 PMC', '2018-10-19')]
+
+        for x in range(len(expected)):
```

Comment:
You mean something like this?

```
            self.assertIsNotNone(entries[x]['uuid'])
            self.assertIsNotNone(entries[x]['updated_on'])
```

Reply:
No exactly. It should be something like: https://github.com/chaoss/grimoirelab-perceval/blob/master/tests/test_git.py#L185 (from line 185 to 218)

Reply:
Got it. I've got the timestamp test working, but not the `uuid`; seems that there's some internal logic related to `uuid`; I guess the code you're looking for is

```
            self.assertEqual(entries[x]['uuid'], uuid(FinosMeetings.metadata_id(entry)))
            self.assertEqual(entries[x]['updated_on'], FinosMeetings.metadata_updated_on(entry))
```

Any idea why the first assert is not working?

Reply:
It's not working because the `uuid` is generated using the `origin` and the value given by `metadata_id`. You can get the `uuid` using the code that I refer above. To be more specific, https://github.com/chaoss/grimoirelab-perceval/blob/master/tests/test_git.py#L195

Reply:
Ha! Clear now, and code fixed:

```
self.assertEqual(entries[x]['uuid'], uuid(finosmeetings.origin, FinosMeetings.metadata_id(entry)))
```